cmd/kubeadm/app/cmd/token.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"text/tabwriter"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	"k8s.io/kubernetes/pkg/api"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/kubectl"
+)
+
+const (
+	bootstrapTokenSecretPrefix = "bootstrap-token-"
+)
+
+func NewCmdToken(out io.Writer) *cobra.Command {
+	tokenCmd := &cobra.Command{
+		Use:   "token",
+		Short: "Manage bootstrap tokens",
+	}
+
+	var tokenDuration string
+	createCmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create discovery tokens on the server.",
+		Run: func(tokenCmd *cobra.Command, args []string) {
+			err := RunCreateToken(out, tokenCmd, tokenDuration)
+			kubeadmutil.CheckErr(err)
+		},
+	}
+	createCmd.PersistentFlags().StringVar(&tokenDuration,
+		"duration", "8h", "Duration the token should be valid")
+	tokenCmd.AddCommand(createCmd)
```

Comment:
I guess we should have an option to specify the token that should be created.
I think something like this is fine
```
# kubeadm create token abcdef.1234567890123456 --duration 8h
abcdef.1234567890123456
# kubeadm create token --duration 8h
fedcba.6543210987654321
```